src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs

```diff
@@ -59,7 +65,7 @@ public override void AdvanceTo(SequencePosition consumed, SequencePosition exami
 
         public override bool TryRead(out ReadResult readResult)
         {
-            TryStart();
+            TryStartAsync();
```

Comment:
Do we expect HTTP/2&3 will also complete this synchronously?

Reply:
Either way, add the asserts to verify?

Reply:
I'm not aware of a reason why we wouldn't expect that - I can add the assert here also

Reply:
Add the assert yeah

Reply:
Done